python/pynessie/commands/contents/commit.py

```diff
@@ -0,0 +1,187 @@
+from ...utils import contents_key, format_key
+
+
+@click.command("commit")
+@click.option("-r", "--ref", help="Branch to commit content to. If not supplied the default branch from config is used.")
+@click.option("-i", "--stdin", is_flag=True, help="Read contents for --set from STDIN (separated by Ctrl-D)")
+@click.option(
+    "-s",
+    "--expect-same-contents",
+    is_flag=True,
+    help="Send the same contents both as the new and expected (old contents) parameters for the edit operations",
+)
+@click.option(
+    "-c",
+    "--condition",
+    "expected_hash",
+    required=True,
+    help="Expected hash. Only perform the action if the branch currently points to the hash specified by this option.",
+)
+@click.option("-d", "--delete", is_flag=True, help="Delete a content.")
+@click.option("-m", "--message", help="Commit message.")
+@click.option("--author", help="The author to use for the commit")
+@click.argument("key", nargs=-1, required=True)
+@pass_client
+@error_handler
+@validate_reference
+def commit(
+    ctx: ContextObject,
+    ref: str,
+    stdin: bool,
+    expect_same_contents: bool,
+    expected_hash: str,
+    delete: bool,
+    message: str,
+    author: str,
+    key: List[str],
+) -> None:
+    """Commit content.
+
+        KEY is the content key that is associated with a specific content to commit.
+    This accepts as well multiple keys with space in between.
+
+    Examples:
+
+        nessie contents commit -m "my awesome commit message" -r dev -c 122345abcd my_table -> Commit to branch 'dev'
+    on expected hash '122345abcd' (head of the branch) the content from the interactive editor with the
+    commit message "my awesome commit message" for content "my_table".
+
+        nessie contents commit -i -r dev -c 122345abcd my_table -> Commit to branch 'dev'
+    on expected hash '122345abcd' (head of the branch) the content from STDIN with the commit message from
+    the interactive editor for content "my_table".
+
+        nessie contents commit -m "my awesome commit message" -d -r dev -c 122345abcd my_table_1 my_table_2 ->
+    Delete contents "my_table_1" and "my_table_2" from branch 'dev' on expected hash '122345abcd' and commit
+    to branch 'dev" with the commit message "my awesome commit message".
+    """
+    operations = _get_commit_operations(ctx.nessie, ref, key, delete, stdin, expect_same_contents)
+
+    if operations:
+        ctx.nessie.commit(ref, expected_hash, _get_message(message, key), author, *operations)
+        click.echo("Changes have been committed.")
+    else:
+        click.echo("No changes requested")
+
+
+def _get_commit_operations(
+    client: NessieClient, ref: str, keys: List[str], is_delete: bool, use_stdin: bool = False, expect_same_contents: bool = False
+) -> List[Operation]:
+    if is_delete:
+        return _get_delete_content_operations(keys)
+
+    return _get_edit_content_operations(client, ref, keys, use_stdin, expect_same_contents)
```

Comment:
IIRC, the old impl. allowed the user to mix edits and deletes in one commit. The user would indicate the wish to delete contents by editing it to an empty file. I think it makes sense to keep that functionality.

Reply:
This was actually a bit confusing for me to mix both functionalities here. I am actually leaning to keep this strictly to edit and the flag to delete the content. At least IMHO, it is pretty clear and what to expect from the command. What is your opinion on this?

Reply:
It is clearer for sure, but functionally it may be important to edit one table and delete another in the _same_ commit. It looks like CLI is losing this capability.

WDYT about using options like `commit --remove key1 --edit key2 --remove key3 --edit key4`?.. and we will not have a generic list of keys, we'll always get them from the options for a specific purpose?

Reply:
for context: the trick with `--expect-same-contents` is when a user manually tweaks a branch that does not have table `A`, for example, but `A` exists on another branch.

Proper `edit` in not possible in this case. Sp, I'm suggesting `--add` to support this use case. It is an edge case, I agree, but it may be critical if/when someone has to "fix" stuff.

Reply:
@dimas-b I am not sure to be honest, the moment when we add more flags, I am afraid this will get complicated 🤔 . Now when I think about it further, maybe the old functionally wasn't bad idea to go around these cases. Do you think would make sense if we just restore it back? Of course keeping as well the remove key flag `--remove-key`. WDYT?

Reply:
SGTM ^ :+1:

Reply:
@dimas-b I have another idea in my head, in case the user input multiple keys, how about we display a list of keys with the operation next to each key and then the user edit the operation, something very similar to git rebase to choose which commits to pick or squash. Not sure how complex to do that in click, maybe worth to look at. WDYT?

Reply:
I like the git-rebase-like approach. Can we go with all exisiting keys by default in case the user does not put any on the cmd. line?

So basically, if the user puts `--remove-key` on the cmd. line, we'll remove all the keys specified, right? But if the user does not put the `--remove-key` option we'll give the user a choice to chose `Put` vs. `Delete` by editing a rebase-like metadata text? Maybe we do not need the `--remove-key` option in this case at all? WDYT?

As far as `click` is concerned, I think we can handle this as editing another temporary file through `click` APIs.

Reply:
Another option is to put a special like after the contents JSON where we have helper data. We could have another like there that selects `Put` vs. `Delete` explicitly. WDYT?

I mean adding this after `# Everything below is ignored`, so we may want to change that like to something like `# Lines starting with #` are ignored (cf. #2392)

Reply:
... but that might be too much for this PR :smile:

Reply:
> So basically, if the user puts `--remove-key` on the cmd. line, we'll remove all the keys specified, right? But if the user does not put the `--remove-key` option we'll give the user a choice to chose `Put` vs. `Delete` by editing a rebase-like metadata text? Maybe we do not need the `--remove-key` option in this case at all? WDYT?

Yes exactly. I think the flag of to remove keys it will make sense IMHO, in case you have like bunch of keys and you just you want to delete them. 

Indeed, I have added a new issue for the git-rebase like feature https://github.com/projectnessie/nessie/issues/2613 . I think this PR is already has many changes and I don't want to add more to it otherwise it will be confusing to follow the changes 😄